src/riscv-constants.h

```diff
@@ -88,28 +102,39 @@ enum MCAUSE_masks : uint64_t {
     MCAUSE_STORE_AMO_ADDRESS_MISALIGNED_MASK = UINT64_C(1) << MCAUSE_STORE_AMO_ADDRESS_MISALIGNED,
     MCAUSE_STORE_AMO_ACCESS_FAULT_MASK = UINT64_C(1) << MCAUSE_STORE_AMO_ACCESS_FAULT,
     MCAUSE_USER_ECALL_MASK = UINT64_C(1) << MCAUSE_USER_ECALL,
-    MCAUSE_SUPERVISOR_ECALL_MASK = UINT64_C(1) << MCAUSE_SUPERVISOR_ECALL,
+    MCAUSE_HSUPERVISOR_ECALL_MASK = UINT64_C(1) << MCAUSE_HSUPERVISOR_ECALL,
+    MCAUSE_VSUPERVISOR_ECALL_MASK = UINT64_C(1) << MCAUSE_VSUPERVISOR_ECALL,
     MCAUSE_MACHINE_ECALL_MASK = UINT64_C(1) << MCAUSE_MACHINE_ECALL,
     MCAUSE_FETCH_PAGE_FAULT_MASK = UINT64_C(1) << MCAUSE_FETCH_PAGE_FAULT,
     MCAUSE_LOAD_PAGE_FAULT_MASK = UINT64_C(1) << MCAUSE_LOAD_PAGE_FAULT,
     MCAUSE_STORE_AMO_PAGE_FAULT_MASK = UINT64_C(1) << MCAUSE_STORE_AMO_PAGE_FAULT
 };
 
-/// \brief medeleg read/write masks
-enum MEDELEG_RW_masks : uint64_t {
-    MEDELEG_W_MASK = MCAUSE_INSN_ADDRESS_MISALIGNED_MASK | MCAUSE_INSN_ACCESS_FAULT_MASK | MCAUSE_ILLEGAL_INSN_MASK |
-        MCAUSE_BREAKPOINT_MASK | MCAUSE_LOAD_ADDRESS_MISALIGNED_MASK | MCAUSE_LOAD_ACCESS_FAULT_MASK |
-        MCAUSE_STORE_AMO_ADDRESS_MISALIGNED_MASK | MCAUSE_STORE_AMO_ACCESS_FAULT_MASK | MCAUSE_USER_ECALL_MASK |
-        MCAUSE_SUPERVISOR_ECALL_MASK | MCAUSE_FETCH_PAGE_FAULT_MASK | MCAUSE_LOAD_PAGE_FAULT_MASK |
-        MCAUSE_STORE_AMO_PAGE_FAULT_MASK
-};
-
 /// \brief Privilege modes
 enum PRV_constants : uint8_t {
-    PRV_U = 0,  ///< User mode
-    PRV_S = 1,  ///< Supervisor mode
-    PRV_HS = 2, ///< Hypervisor-extended supervisor mode
-    PRV_M = 3   ///< Machine mode
+    PRV_U = 0,     ///< User mode
+    PRV_HS = 1,    ///< Supervisor mode
+    PRV_RSVD = 2,  ///< Reserved
+    PRV_M = 3      ///< Machine mode
+};
+
+/// \brief Access mode shifts
+enum ACCESS_MODE_SHIFTS : uint8_t {
+    VIRTUAL_BIT_SHIFT = 7
```

Comment:
I think a better name is `ACCESS_MODE_VRT_SHIFT`. And you should also add `ACCESS_MODE_PRV_SHIFT = 0`.